plugins/commands/snapshot/command/push_shared.rb

```diff
@@ -0,0 +1,56 @@
+require 'json'
+
+module VagrantPlugins
+  module CommandSnapshot
+    module Command
+      module PushShared
+        def shared_exec(argv, m)
+          with_target_vms(argv) do |vm|
+            if !vm.id
+              vm.ui.info("Not created. Cannot push snapshot state.")
+              next
+            end
+
+            vm.env.lock("machine-snapshot-stack") do
+              m.call(vm)
+            end
+          end
+
+          0
+        end
+
+        def push(machine)
+          snapshot_name = "push_#{Time.now.to_i}_#{rand(10000)}"
```

Comment:
Could we interpolate something meaningful from the machine here, like the machine id or machine name? I worry about a user going into the GUI or file system and seeing all these snapshots and not knowing what is what.